crates/uv-fs/src/lib.rs

```diff
@@ -19,13 +19,22 @@ mod path;
 pub async fn read_to_string(path: impl AsRef<Path>) -> std::io::Result<String> {
```

Comment:
Are there call sites to this function for which we may not want to accept UTF-16? 

I suggest to at least document that this function supports UTF16 decoding.

Reply:
Yeah, I'll probably split this into two functions so that callers can specifically opt into it.

Reply:
I split it into two functions. This was also being used to read `pyproject.toml`, and it's not obvious to me that we should support UTF-16 there. So that behavior remains unchanged.